app/middlewares/auth.ts

```diff
@@ -16,6 +18,14 @@ export async function secure(req: Request, res: Response, next: NextFunction): P
       return next();
     }
   }
-
   return next(authenticationError('Error: Unauthorized'));
 }
+
+export function adminRole(req: Request, res: Response, next: NextFunction): void {
+  if (req.user && req.user.role === ROLES.ADMIN) {
```

Comment:
I recommend that you use: [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html)